src/runner.js

```diff
@@ -89,6 +92,17 @@ assign(Runner.prototype, {
         stream.emit('error', err);
       }
       return runner.client.stream(runner.connection, sql, stream, options);
+    }).catch(function(err) {
```

Comment:
if this catch would be in the same position where there was `.catch(noop)` earlier, then the `if (hasHandler) throw err;` line could be removed.